### ISSUE TITLE
Generate BigQuery Dataset Definitions

### ISSUE THREAD

**Issue by gmcfall** [konigio/konig#164]
As a data architect, I want a tool that will generate BigQuery Dataset definitions so that I can deploy the dataset(s) for my project to Google Cloud Platform.

### Acceptance Criteria

The tool must generate a default Dataset definition for each dataset referenced by a BigQuery table.
The  definition must have the following form:

```
{
   "datasetReference" : {
      "datasetId" : string,
      "projectId" : string
   }
}
```

The value of the `datasetId` and `projectId` properties must be derived from the generated table definitions.

In other words, the tool must scan all generated table definitions and create a data set definition for each table found.  These small JSON documents should be stored in files that match the following template:

```
   {bigQueryBaseDir}/dataset/{datasetId}.json
```
